crates/re_renderer/src/renderer/point_cloud.rs

```diff
@@ -320,32 +339,73 @@ impl PointCloudDrawData {
+                    batches
+                        .iter()
+                        .flat_map(|batch_info| {
+                            batch_info
+                                .additional_outline_mask_ids_vertex_ranges
+                                .iter()
+                                .map(|(_, mask)| gpu_data::BatchUniformBuffer {
+                                    world_from_obj: batch_info.world_from_obj.into(),
+                                    flags: batch_info.flags.bits.into(),
+                                    outline_mask_ids: mask.0.unwrap_or_default().into(),
+                                    end_padding: Default::default(),
+                                })
+                        })
+                        .collect::<Vec<_>>()
+                        .into_iter(),
+                )
+                .into_iter();
+
             let mut start_point_for_next_batch = 0;
             for (batch_info, uniform_buffer_binding) in
                 batches.iter().zip(uniform_buffer_bindings.into_iter())
             {
-                let bind_group = ctx.gpu_resources.bind_groups.alloc(
-                    &ctx.device,
-                    &ctx.gpu_resources,
-                    &BindGroupDesc {
-                        label: batch_info.label.clone(),
-                        entries: smallvec![uniform_buffer_binding],
-                        layout: point_renderer.bind_group_layout_batch,
-                    },
-                );
-
                 let point_vertex_range_end = (start_point_for_next_batch + batch_info.point_count)
                     .min(Self::MAX_NUM_POINTS as u32);
+                let mut active_phases = enum_set![DrawPhase::Opaque];
+                // Does the entire batch participate in the outline mask phase?
+                if batch_info.overall_outline_mask_ids.is_some() {
+                    active_phases.insert(DrawPhase::OutlineMask);
+                }
 
-                batches_internal.push(PointCloudBatch {
-                    bind_group,
-                    vertex_range: (start_point_for_next_batch * 6)
-                        ..((start_point_for_next_batch + batch_info.point_count) * 6),
-                });
+                batches_internal.push(point_renderer.create_point_cloud_batch(
+                    ctx,
+                    batch_info.label.clone(),
+                    uniform_buffer_binding,
+                    start_point_for_next_batch..point_vertex_range_end,
+                    active_phases,
+                ));
+
+                for (range, _) in &batch_info.additional_outline_mask_ids_vertex_ranges {
+                    batches_internal.push(
+                        point_renderer.create_point_cloud_batch(
+                            ctx,
+                            batch_info
+                                .label
+                                .clone()
+                                .push_str(&format!("strip-only {range:?}")),
```

Comment:
Looks like this is missing a space. Could also be written as

```suggestion
                            format!("{} strip-only {:?}",  batch_info.label, range),
```